propertyestimator/utils/quantities.py

```diff
@@ -0,0 +1,656 @@
+
+    def __init__(self, value, uncertainty):
+        """Constructs a new EstimatedQuantity object.
+
+        Parameters
+        ----------
+        value: unit.Quantity
+            The value of the estimated quantity.
+        uncertainty: unit.Quantity
+            The uncertainty in the value of the estimated quantity.
+        """
+
+        assert value is not None and uncertainty is not None
+
+        assert isinstance(value, unit.Quantity)
+        assert isinstance(uncertainty, unit.Quantity)
+
+        assert value.unit.is_compatible(uncertainty.unit)
+
+        self._value = value
+        self._uncertainty = uncertainty
+
+    def __add__(self, other):
+
+        assert isinstance(other, VanillaEstimatedQuantity)
+
+        self_ufloat, self_unit = VanillaEstimatedQuantity._get_uncertainty_object(self)
+        other_ufloat, other_unit = VanillaEstimatedQuantity._get_uncertainty_object(other)
+
+        assert self_unit == other_unit
+
+        result_ufloat = self_ufloat + other_ufloat
+
+        result_value = result_ufloat.nominal_value * self_unit
+        result_uncertainty = result_ufloat.std_dev * self_unit
+
+        return VanillaEstimatedQuantity(result_value, result_uncertainty)
+
+    def __sub__(self, other):
+
+        assert isinstance(other, VanillaEstimatedQuantity)
+
+        self_ufloat, self_unit = VanillaEstimatedQuantity._get_uncertainty_object(self)
+        other_ufloat, other_unit = VanillaEstimatedQuantity._get_uncertainty_object(other)
+
+        assert self_unit == other_unit
+
+        result_ufloat = self_ufloat - other_ufloat
+
+        result_value = result_ufloat.nominal_value * self_unit
+        result_uncertainty = result_ufloat.std_dev * self_unit
+
+        return VanillaEstimatedQuantity(result_value, result_uncertainty)
+
+    def __mul__(self, other):
+
+        # We only support multiplication by a scalar here.
+        assert np.issubdtype(type(other), float) or np.issubdtype(type(other), int)
+
+        self_ufloat, self_unit = VanillaEstimatedQuantity._get_uncertainty_object(self)
+
+        result_ufloat = self_ufloat * other
+
+        result_value = result_ufloat.nominal_value * self_unit
+        result_uncertainty = result_ufloat.std_dev * self_unit
+
+        return VanillaEstimatedQuantity(result_value, result_uncertainty)
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+
+        self_ufloat, self_unit = VanillaEstimatedQuantity._get_uncertainty_object(self)
+
+        result_ufloat = self_ufloat / other
+
+        result_value = result_ufloat.nominal_value * self_unit
+        result_uncertainty = result_ufloat.std_dev * self_unit
+
+        return VanillaEstimatedQuantity(result_value, result_uncertainty)
+
+    def __eq__(self, other):
+        if not isinstance(other, VanillaEstimatedQuantity):
+            return False
+
+        return self.value == other.value and self.uncertainty == other.uncertainty
```

Comment:
I guess this is a conservative implementation, but we probably need to be careful when defining equality in a statistical sense. This criterion will probably never be true unless the second quantity is a copy of the first. Using `np.isclose` may be unexpected behavior. I'm thinking that for equality checking, the safe option we be to add explicit methods rather than Python operators and raise an exception declaring that equating two statistical quantities is ambiguous (similar to `np.ndarray`) here.

Reply:
>I'm thinking that for equality checking, the safe option we be to add explicit methods rather than Python operators and raise an exception declaring that equating two statistical quantities is ambiguous (similar to `np.ndarray`) here.

I agree that is the safer (but less convenient) option here, maybe also for `>`, `<`, ... operators.

Note that the `uncertainties` package interprets comparison operators in a specific way, described in [section 4.3 of the tech guide](https://media.readthedocs.org/pdf/uncertainties-python-package/latest/uncertainties-python-package.pdf#page=26&zoom=100,0,96).